Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -152,7 +152,8 @@ bool loadCanceled
     private DafnyDocument VerifyInternal(VerifyRequest verifyRequest) {
       var (document, cancellationToken) = verifyRequest;
       notificationPublisher.SendStatusNotification(document.Text, CompilationStatus.VerificationStarted);
-      var verificationResult = verifier.Verify(document.Program, cancellationToken);
+      var progressReporter = new VerificationProgressReporter(document.Text, notificationPublisher);
```

Comment:
I'd prefer that the instance of `VerificationProgressReporter` is provided through dependency injection. See [LanguageServerExtensions](https://github.com/dafny-lang/dafny/blob/master/Source/DafnyLanguageServer/Workspace/LanguageServerExtensions.cs). Then, the actual instance can be injected through the constructor of `DafnyProgramVerifier`.
However, this will require that `VerificationProgressReporter` will be a singleton instance; thus, the `reportProgress` method has to receive the actual document.

Reply:
Would that really be the right choice here? VerificationProgressReporter is a purely internal wapper around notificationPublisher to connect the Boogie API to the notifications API: what are cases at this point of the code where we would want to pass in something else than an instance of this specific VerificationErrorReporter? (Originally I thought I wouldn't even have that interface at all, but I introduced it to decouple the two APIs.

In any case I've changed `VerificationErrorReporter` to be a private class, which is certainly better.  Let me know if you think I should change it further.